src/sentry/notifications/platform/slack/provider.py

```diff
@@ -69,11 +74,43 @@ class SlackNotificationProvider(NotificationProvider[SlackRenderable]):
         NotificationTargetResourceType.DIRECT_MESSAGE,
     ]
 
+    @classmethod
+    def validate_target(cls, *, target: NotificationTarget) -> None:
+        super().validate_target(target=target)
+
+        assert isinstance(
+            target, cls.target_class
+        ), "Target for SlackNotificationProvider must be a IntegrationNotificationTarget"
+
+        # 1. Validate the integration exists
+        # 2. Validate the organization integration exists
+        validate_integration_for_target(target=target)
+
+        # 3. Validate the Slack channel or user exists
+        channel_name = target.specific_data["channel_name"]
+        try:
+            validate_slack_entity_id(
+                integration_id=target.integration_id,
+                input_name=channel_name,
+                input_id=target.resource_id,
+            )
+        except (ValidationError, IntegrationError) as e:
+            raise NotificationProviderError(
+                f"Slack channel or user with id '{target.resource_id}' could not be validated"
+            ) from e
+
```

Comment:
I don't actually think we need this, this will actually make an API call to validate the channel ID against the name every time, when the notification platform doesn't really care that, I think it's okay to attempt the call and fail.

Also it'd require the requestor to always attach a channel name, when in reality we only need an ID so I don't think it's necessary.